Custom query fix

mhavey opened the pull request:
Pull request for RDF connector changes

abhishekpradeepmishra committed:
code merge, custom query changes

abhishekpradeepmishra committed:
documentation update for custom query

mhavey committed:
RDF

burhan94 committed:
Merge branch 'master' into custom-query

mhavey committed:
simplify

mhavey committed:
Update README.md

mhavey committed:
Create PropertyGraph.md

mhavey committed:
Create RDF.md

mhavey committed:
Update README.md

mhavey committed:
Update PropertyGraph.md

mhavey committed:
Update README.md

mhavey committed:
Update PropertyGraph.md

mhavey committed:
Update PropertyGraph.md

mhavey committed:
Update RDF.md

mhavey committed:
Create sample-cli-rdf.sh

mhavey committed:
Update sample-cli-script.sh

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Update README.md

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

mhavey committed:
Add files via upload

mhavey committed:
Update RDF.md

mhavey committed:
Update RDF.md

abhishekpradeepmishra commented on athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java:
remove this

//Client client = null; NOT USED

mhavey replied:
agreed

abhishekpradeepmishra commented on athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java:
```diff
@@ -30,7 +30,7 @@ public class NeptuneConnection
     private static Cluster cluster = null;
 
-    NeptuneConnection(String neptuneEndpoint, String neptunePort, boolean enabledIAM)
+    protected NeptuneConnection(String neptuneEndpoint, String neptunePort, boolean enabledIAM, String region)
```
where is the region parameter used?

mhavey replied:
sparql connection needs region

abhishekpradeepmishra commented on athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java:
```diff
@@ -0,0 +1,221 @@
+    {
+        super(neptuneEndpoint, neptunePort, enabledIAM, region);
+        this.connectionString = "https://" + neptuneEndpoint + ":" + neptunePort;
+        this.enabledIAM = enabledIAM;
+        this.region = region;
```
won't this be available from base class

mhavey replied:
modified

abhishekpradeepmishra commented on athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java:
```diff
@@ -0,0 +1,221 @@
+            try {
+                this.neptuneSparqlRepo = new NeptuneSparqlRepository(this.connectionString, awsCredentialsProvider, this.region);
+            } 
+            catch (NeptuneSigV4SignerException e) {
+                e.printStackTrace();
+                logger.error("Error connecting with IAM auth", e);
+            }
```
Don't catch the exception here. Move all exception catch code to NeptuneRecordHandler

mhavey replied:
I will keep some catch blocks -- for the safeClose() methods. But this one I will bubble up

abhishekpradeepmishra commented on athena-neptune/docs/aws-glue-sample-scripts/RDF.md:
```diff
@@ -0,0 +1,169 @@
+|componenttype|rdf||
+|querymode|class, sparql||
+|sparql|SPARQL query to use to find resultset.|Only if querymode='sparql'. Omit prefixes. Define prefixes as table properties.|
+|classuri|Class of resources to find|In curie form prefix:classname. Only if querymode='class'. Connector will query for resources whose RDF type is this classuri.|
```
is this intended spelling?

abhishekpradeepmishra replied:
mark config options like preds_prefix as optional

abhishekpradeepmishra commented on athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/RDFHandler.java:
```diff
@@ -0,0 +1,183 @@
+        }
+
+        // 3. Add filters. This means adding FILTER conditions to the SPARQL.
+        // No longer used
```
remove this

mhavey replied:
agreed

abhishekpradeepmishra commented:
I have added comments, kindly review them.

I will test the connector next week

abhishekpradeepmishra committed:
updated table creation script

abhishekpradeepmishra commented on athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh:
```diff
@@ -0,0 +1,77 @@
+
+echo $1;
+echo $2;
+echo $3;
```
remove the 3rd parameter, its not needed and update the script accordingly

mhavey replied:
removed

abhishekpradeepmishra commented on athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh:
```diff
@@ -0,0 +1,77 @@
+        } 
+        }' \
+    --profile $1 \
+    --endpoint https://glue.$2.amazonaws.com
```
missing back slash, script breaks

mhavey replied:
done

abhishekpradeepmishra commented on athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh:
```diff
@@ -0,0 +1,77 @@
+        } 
+        }' \
+    --profile $1 \
+    --endpoint https://glue.$2.amazonaws.com
```
missing back slash, script breaks

abhishekpradeepmishra replied:
'--table-input': Invalid JSON: Expecting property name enclosed in double quotes: line 16 column 9 (char 813)
JSON received: {"Name":"airport_route", "StorageDescriptor":{ 
        "Columns":[

mhavey replied:
I'll test it

abhishekpradeepmishra commented on athena-neptune/docs/aws-glue-sample-scripts/RDF.md:
run spell check on the file

abhishekpradeepmishra commented on athena-neptune/athena-neptune.yaml:
```diff
@@ -25,7 +25,7 @@ Parameters:
     Type: String
   NeptuneGraphType:
-    Description: 'Type of graph created in Neptune, defaults to PROPERTYGRAPH. RDF support is yet to be implemented'
+    Description: 'Type of graph created in Neptune, defaults to PROPERTYGRAPH. Allowed values: PROPERTYGRAPH, RDF'
```
add this as allowed list values

"AllowedValues" : ["PROPERTYGRAPH", "RDF"],

mhavey replied:
added

mhavey committed:
Update sample-cli-rdf.sh

mhavey committed:
Update RDF.md

mhavey committed:
Add files via upload

mhavey committed:
Update RDF.md

mhavey committed:
Update athena-neptune.yaml

mhavey committed:
Update NeptuneRecordHandler.java

mhavey committed:
Update NeptuneConnection.java

mhavey committed:
Update RDFHandler.java

mhavey committed:
Update NeptuneSparqlConnection.java

abhishekpradeepmishra committed:
changed componenttype to view, and other related changes

abhishekpradeepmishra committed:
Merge branch 'awslabs:master' into custom-query

abhishekpradeepmishra committed:
Merge pull request #11 from abhishekpradeepmishra/custom-query

Custom query change as per review

mhavey committed:
Update sample-cli-script.sh

mhavey committed:
loglevel

mhavey committed:
loglevel

dependabot[bot] committed:
build(deps): bump aws-sdk.version from 1.12.514 to 1.12.533 (#1426)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.ivy:ivy from 2.5.1 to 2.5.2 in /athena-neptune (#1427)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

akuzin1 committed:
Fix case insensitive table name search bug (#1429)

dependabot[bot] committed:
build(deps): bump netty.version from 4.1.93.Final to 4.1.97.Final (#1430)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump software.amazon.msk:aws-msk-iam-auth from 1.1.7 to 1.1.9 (#1425)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.protobuf:protobuf-java from 3.23.4 to 3.24.1 (#1423)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.junit:junit-bom from 5.9.3 to 5.10.0 (#1392)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.zookeeper:zookeeper from 3.8.1 to 3.8.2 (#1391)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: chngpe <102991671+chngpe@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump software.amazon.jsii:jsii-runtime from 1.84.0 to 1.88.0 (#1432)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.amazon.redshift:redshift-jdbc42 from 2.1.0.17 to 2.1.0.18 (#1408)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.microsoft.azure:msal4j from 1.13.9 to 1.13.10 (#1420)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump gremlinDriverVersion from 3.6.4 to 3.7.0 (#1410)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.commons:commons-lang3 from 3.12.0 to 3.13.0 (#1406)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.sap.cloud.db.jdbc:ngdbc from 2.17.10 to 2.17.12 (#1407)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.zookeeper:zookeeper from 3.8.1 to 3.9.0 (#1405)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.yaml:snakeyaml from 2.0 to 2.1 (#1413)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump nl.jqno.equalsverifier:equalsverifier from 3.15 to 3.15.1 (#1412)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aws-cdk-lib from 2.63.2 to 2.80.0 in /validation_testing/cdk_federation_infra_provisioning/app (#1424)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-storage from 2.24.0 to 2.26.1 (#1421)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.junit.support:testng-engine from 1.0.1 to 1.0.4 (#1409)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

akuzin1 committed:
Revert "build(deps): bump aws-cdk-lib from 2.63.2 to 2.80.0 in /validation_testing/cdk_federation_infra_provisioning/app" (#1433)

henrymai committed:
Use Arrow 13 (#1447)

akuzin1 committed:
Revert "Use Arrow 13 (#1447)"

This reverts commit 897d787f75df22861434e82b3cebd843bd1923c2.

chngpe committed:
Fix dynamic filter date on timezone issue. (#1453)

akuzin1 committed:
Revert revert arrow13 (#1471)

dependabot[bot] committed:
build(deps): bump org.eclipse.jetty:jetty-xml from 11.0.15 to 11.0.16 in /athena-hbase (#1472)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kparwal committed:
Transition to AES-GCM for UDFs (#708)

Co-authored-by: Abdul Al-Faraj <ar.al-faraj@outlook.com>
Co-authored-by: AbdulRehman Faraj <arfaraj@amazon.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-bigquery from 2.30.1 to 2.31.2 (#1467)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aws-sdk.version from 1.12.533 to 1.12.547 (#1465)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump net.jqwik:jqwik from 1.7.4 to 1.8.0 (#1461)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.commons:commons-lang3 from 3.12.0 to 3.13.0 (#1460)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.guava:guava from 32.1.1-jre to 32.1.2-jre (#1444)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aws-sdk.version from 1.12.533 to 1.12.548 (#1474)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.microsoft.sqlserver:mssql-jdbc from 11.2.3.jre11 to 12.4.1.jre11 (#1459)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump net.sf.jt400:jt400 from 20.0.0 to 20.0.2 (#1436)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-resourcemanager from 1.23.0 to 1.25.0 (#1435)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump junit:junit from 4.13.1 to 4.13.2 (#1439)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.yaml:snakeyaml from 2.1 to 2.2 (#1441)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.protobuf:protobuf-java from 3.24.1 to 3.24.3 (#1469)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.directory.api:api-ldap-model from 2.1.3 to 2.1.4 (#1458)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump slf4j-log4j.version from 2.0.7 to 2.0.9 (#1455)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.amazonaws:aws-lambda-java-core from 1.2.2 to 1.2.3 (#1446)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump net.snowflake:snowflake-jdbc from 3.13.33 to 3.14.1 (#1437)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump io.lettuce:lettuce-core from 6.2.5.RELEASE to 6.2.6.RELEASE (#1440)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.api.grpc:grpc-google-cloud-bigquerystorage-v1 from 2.41.0 to 2.41.1 (#1443)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

akuzin1 committed:
Modified JDBC connectors to fetch metadata from underlying system tables. (#1463)

dependabot[bot] committed:
build(deps): bump maven-shade-plugin from 3.4.1 to 3.5.0 (#1356)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Abdul Al-Faraj <ar.al-faraj@outlook.com>

dependabot[bot] committed:
build(deps): bump actions/checkout from 3 to 4 (#1470)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Abdul Al-Faraj <ar.al-faraj@outlook.com>

akuzin1 committed:
Excluding large Mac/Windows Statically compiled files (#1475)

Co-authored-by: AbdulRehman Faraj <arfaraj@amazon.com>
Co-authored-by: Abdul Al-Faraj <ar.al-faraj@outlook.com>

chngpe committed:
temporary disable and/or complex expression (#1486)

chngpe committed:
DocDB skip collections that user has no permission (#1491)

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-resourcemanager from 1.25.0 to 1.26.0 (#1484)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-bigquery from 2.31.2 to 2.32.0 (#1483)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.amazon.redshift:redshift-jdbc42 from 2.1.0.18 to 2.1.0.19 (#1482)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.api.grpc:grpc-google-cloud-bigquerystorage-v1 from 2.41.1 to 2.42.0 (#1480)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aws-sdk.version from 1.12.548 to 1.12.552 (#1477)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-storage from 2.26.1 to 2.27.0 (#1476)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump Impala:ImpalaJDBC42 from 2.6.30.1036 to 2.6.32.1041 (#1478)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump Hive:HiveJDBC42 from 2.6.15.1018 to 2.6.21.1025 (#1479)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump software.amazon.jsii:jsii-runtime from 1.88.0 to 1.89.0 (#1495)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Abdul Al-Faraj <ar.al-faraj@outlook.com>

dependabot[bot] committed:
build(deps): bump org.apache.maven.plugins:maven-shade-plugin from 3.5.0 to 3.5.1 (#1496)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.maven.plugins:maven-javadoc-plugin from 3.5.0 to 3.6.0 (#1481)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.avro:avro from 1.11.2 to 1.11.3 (#1500)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump net.sf.jt400:jt400 from 20.0.2 to 20.0.3 (#1498)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump nl.jqno.equalsverifier:equalsverifier from 3.15.1 to 3.15.2 (#1499)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-bigquery from 2.32.0 to 2.33.1 (#1507)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Abdul Al-Faraj <ar.al-faraj@outlook.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-storage from 2.27.0 to 2.27.1 (#1505)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.api.grpc:grpc-google-cloud-bigquerystorage-v1 from 2.42.0 to 2.43.0 (#1506)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump netty.version from 4.1.97.Final to 4.1.99.Final (#1504)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aws-sdk.version from 1.12.552 to 1.12.560 (#1503)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Abdul Al-Faraj <ar.al-faraj@outlook.com>

ejeffrli committed:
Implement validation tests with PostgreSQL and Redshift.
Add spill bucket to workflow

Add redshift to readme

mhavey committed:
PR review

mhavey committed:
fixed issue in record handler

chngpe committed:
predicate pushdown fix on complex expression (#1508)

chngpe committed:
Merge branch 'master' into validation_testing_integrations

ejeffrli committed:
Validation testing integrations (#1509)

ejeffrli committed:
Add case insensitive schema functionality

Add postgresql and mysql case insensitive functionality and mysql tests

Add redshift case insesnsitive functionality and tests

Set enable_case_sensitive_identifier in redshift cluster group and fix etl case insensitive job

Specify customer and customer_address schema case

Fix unit tests to work with case insensitive schema implementation

Allow unit tests to work with case insensitive schema

Refactor redshift case insensitive logic

Remove debugging messages

Add materialized view tests and refactor code to remove duplicate redshift and postgres code

Remove duplicate check

akuzin1 committed:
Fixing typo for look up of Materialized View in Postgres connector  (#1521)

dependabot[bot] committed:
build(deps): bump org.apache.zookeeper:zookeeper from 3.9.0 to 3.9.1 (#1511)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump net.snowflake:snowflake-jdbc from 3.14.1 to 3.14.2 (#1512)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump software.amazon.jsii:jsii-runtime from 1.89.0 to 1.90.0 (#1515)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.oracle.database.jdbc:ojdbc8 from 23.2.0.0 to 23.3.0.23.09 (#1514)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.sap.cloud.db.jdbc:ngdbc from 2.17.12 to 2.18.13 (#1513)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump Hive:HiveJDBC42 from 2.6.21.1025 to 2.6.23.1027 (#1516)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.protobuf:protobuf-java from 3.24.3 to 3.24.4 (#1517)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.kafka:kafka-clients from 3.5.1 to 3.6.0 (#1518)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump netty.version from 4.1.99.Final to 4.1.100.Final (#1525)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aws-sdk.version from 1.12.560 to 1.12.566 (#1524)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump fasterxml.jackson.version from 2.15.2 to 2.15.3 (#1523)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-resourcemanager from 1.26.0 to 1.30.0 (#1526)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.github.spotbugs:spotbugs-annotations from 4.7.3 to 4.8.0 (#1527)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-bigquery from 2.33.1 to 2.33.2 (#1528)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.directory.api:api-ldap-model from 2.1.4 to 2.1.5 (#1530)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-storage from 2.27.1 to 2.28.0 (#1531)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.api.grpc:grpc-google-cloud-bigquerystorage-v1 from 2.43.0 to 2.44.0 (#1532)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.guava:guava from 32.1.2-jre to 32.1.3-jre (#1534)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

akuzin1 committed:
Debug release workflow (#1536)

chngpe committed:
add license header for BigQueryFederationExpressionParserTest (#1537)

akuzin1 committed:
Modified maven_push workflow (#1538)

ejeffrli committed:
Add case insensitive schema functionality (#1539)

mhavey committed:
merge

ejeffrli committed:
Fix checksytle error in JDBCUtil

ejeffrli committed:
Fix checkstyle error in JDBCUtil (#1551)

ejeffrli committed:
Add Mysql pagination implementation

Add postgresql and redshift pagination implementations

Clean up code and wrote more informative logs and comments

Remove duplicate code in individual connectors and add pagination unit tests

Mocked paginated result sets for more code coverage

Revert pom.xml dev changes

Add more informative logs

Remove semicolon

burhan94 committed:
Merge branch 'master' into custom-query-fix

ejeffrli committed:
Add missing check to template with KMS keys

ejeffrli committed:
Add missing check to template with KMS keys (#1572)

burhan94 committed:
Merge branch 'master' into custom-query-fix

ejeffrli committed:
Implement MySQL, PostgreSQL, Redshift Connectors' pagination (#1567)

ejeffrli committed:
Implement dynamodb pagination (#1568)

mhavey committed:
Update PropertyGraphHandler.java

fix import

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-storage from 2.28.0 to 2.29.1 (#1571)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump fkirc/skip-duplicate-actions from 5.3.0 to 5.3.1 (#1541)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ejeffrli committed:
Implement elasticsearch pagination (#1581)

dependabot[bot] committed:
build(deps): bump commons-cli:commons-cli from 1.5.0 to 1.6.0 (#1555)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: ejeffrli <144148373+ejeffrli@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.directory.server:apacheds-kerberos-codec from 2.0.0.AM26 to 2.0.0.AM27 (#1549)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.jacoco:jacoco-maven-plugin from 0.8.10 to 0.8.11 (#1550)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.microsoft.sqlserver:mssql-jdbc from 12.4.1.jre11 to 12.4.2.jre11 (#1554)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.amazon.redshift:redshift-jdbc42 from 2.1.0.19 to 2.1.0.22 (#1578)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-bigquery from 2.33.2 to 2.34.2 (#1580)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: chngpe <102991671+chngpe@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.microsoft.azure:msal4j from 1.13.10 to 1.14.0 (#1564)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.mysql:mysql-connector-j from 8.1.0 to 8.2.0 (#1559)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump com.google.cloud:google-cloud-resourcemanager from 1.30.0 to 1.32.0 (#1579)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.maven.plugins:maven-checkstyle-plugin from 3.3.0 to 3.3.1 (#1563)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aws-sdk.version from 1.12.566 to 1.12.589 (#1584)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: ejeffrli <144148373+ejeffrli@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.codehaus.mojo:license-maven-plugin from 2.2.0 to 2.3.0 (#1566)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: ejeffrli <144148373+ejeffrli@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump org.apache.maven.plugins:maven-dependency-plugin from 3.6.0 to 3.6.1 (#1562)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: ejeffrli <144148373+ejeffrli@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump log4j2Version from 2.20.0 to 2.21.1 (#1558)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/setup-node from 3 to 4 (#1552)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump hbase.version from 2.5.5-hadoop3 to 2.5.6-hadoop3 (#1544)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump net.jqwik:jqwik from 1.8.0 to 1.8.1 (#1543)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: ejeffrli <144148373+ejeffrli@users.noreply.github.com>

mhavey committed:
Merge branch 'master' into custom-query-fix

ejeffrli committed:
Update postgreSQL version to 13.7 (#1585)

mhavey committed:
Merge branch 'master' into custom-query-fix

chngpe committed:
Update google cloud bigquerystorage version to 2.46 (#1589)

AbdulR3hman committed:
Support Redshift External Schemas & Tables (#1610)

Co-authored-by: AbdulRehman Faraj <arfaraj@amazon.com>

mhavey committed:
Merge branch 'master' into custom-query-fix

mhavey committed:
Update README.md - remove merge comments

mhavey committed:
Update README.md - remove merge comments
